src/WebJobs.Script/Diagnostics/ILoggerFactoryBuilder.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// Provides ways to plug into the ScriptHost ILoggerFactory initialization.
+    /// </summary>
+    public interface ILoggerFactoryBuilder
```

Comment:
Also, you want this to be a core SDK capability, not specific to Functions right? I.e. whatever extensibility you add here should be in core SDK?